### PR TITLE
Aria improvements

### PR DESCRIPTION
**Pull request by schanzer**
This PR contains a number of changes, designed to make blocks more useful with screen readers:

- AST nodes now hold pointers to their siblings, parents and first child
- Arrow keys now navigating between generations (up/down) and siblings (left/right)
- Tab and Shit-Tab are preserved -- _should they be?_
- `aria-label`s in the WeScheme Parser are now decoded to properly sound out `?`, `!`, `*`, etc
- An `ariaError` node has been added to the CM wrapper, which we use to read alerts. For now, this is just error messages, but I could also see us using this for other things down the road.

### CODE REVIEW

File: spec/blocks-test.js [hunk 1]
```diff
@@ -370,6 +377,67 @@ describe('The CodeMirrorBlocks Class', function() {
...
+          this.firstArg = this.blocks.ast.rootNodes[0].args[0];
+          this.secondArg = this.blocks.ast.rootNodes[0].args[1];
+          this.thirdArg = this.blocks.ast.rootNodes[0].args[2];
+        });
+
+        it('left-arrow should navigate to the next sibling, but not beyond it', function() {
+          this.secondRoot.el.dispatchEvent(click());
+          expect(document.activeElement).toBe(this.secondRoot.el);
+          expect(this.blocks.scroller.getAttribute('aria-activedescendent')).toBe(this.secondRoot.el.id);
+          this.firstRoot.el.dispatchEvent(keydown(37));
```
Comment (pcardune) [inline spec/blocks-test.js]: you could put these key codes in a const to make it easier to understand what is going on.

File: src/ast.js [hunk 2]
```diff
@@ -85,14 +118,13 @@ export class Unknown extends ASTNode {
...
+    setNavigationPointers(this.elts, this);
   }
 
   *[Symbol.iterator]() {
     yield this;
     for (let elt of this.elts) {
-      for (let node of elt) {
-        yield node;
-      }
+      yield* elt;
```
Comment (pcardune) [inline src/ast.js]: nice

File: src/blocks.js [hunk 3]
```diff
@@ -551,28 +551,47 @@ export default class CodeMirrorBlocks {
...
...EEECUBYln03TTTdNBDZopopYvrTTdNa325mImNg3TTPV9q3pmY0xoO6bv3r00y+IDGid/9aaaZTGMuj9mpu9Mpio1dXrr5HERTZSmqU36A3CumzN/9Robv/Xx4v9ijkSRSNLQhAWumap82WRSBUqXStV/YcS+XVLnSS+WLDroqArFkMEsAS+eWmrUzrO0oEmE40RlMZ5+ODIkAyKAGUwZ3mVKmcamcJnMW26MRPgUw6j+LkhyHGVGYjSUUKNpuJUQoOIAyDvEyG8S5yfK6dhZc0Tx1KI/gviKL6qvvFs1+bWtaz58uUNnryq6kt5RzOCkPWlVqVX2a/EEBUdU1KrXLf40GoiiFXK///qpoiDXrOgqDR38JB0bw7SoL+ZB9o1RCkQjQ2CBYZKd/+VJxZRRZlqSkKiws0WFxUyCwsKiMy7hUVFhIaCrNQsKkTIsLivwKKigsj8XYlwt/WKi2N4d//uQRCSAAjURNIHpMZBGYiaQPSYyAAABLAAAAAAAACWAAAAApUF/Mg+0aohSIRobBAsMlO//Kk4soosy1JSFRYWaLC4qZBYWFRGZdwqKiwkNBVmoWFSJkWFxX4FFRQWR+LsS4W/rFRb/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////VEFHAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAU291bmRib3kuZGUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMjAwNGh0dHA6Ly93d3cuc291bmRib3kuZGUAAAAAAAAAACU=");
```
Comment (pcardune) [inline src/blocks.js]: you should be able to put this in a ".wav" file and just import the wav file. webpack will magically inline the data url for you. You may need to update the webpack config. Just look up what it does for png files.

File: src/languages/wescheme/WeschemeParser.js [hunk 4]
```diff
@@ -21,12 +21,43 @@ try {
...
+    str = str.toString();
   }
-  return aria;
+
+  if(symbolMap.get(str)) { 
+    // translate simple symbols
+    return symbolMap.get(str);
+  } else {
+    // pronounce special chars, scheme-style
+    str = str.replace("?", " huh").replace("!"," bang");
```
Comment (pcardune) [inline src/languages/wescheme/WeschemeParser.js]: huh, that's funny.

File: src/languages/wescheme/WeschemeParser.js [hunk 5]
```diff
@@ -21,12 +21,43 @@ try {
...
+  }
+}
+
+// expressionAria: String Array -> String
+function expressionAria(func, args) {
+  return symbolAria(func) + ' expression, ' + pluralize('argument', args);
+}
+
+function pluralize(noun, set) {
+  return set.length + ' ' + noun + (set.length!=1? 's' : '');
```
Comment (pcardune) [inline src/languages/wescheme/WeschemeParser.js]: oh the whitespace, it hurts.

**Comment by pcardune** [comment]
generally looks good to me. Mostly just styling comments. I will try to update the lint config a bit more, but feel free to commit this as is, or respond to my comments. Probably the main thing I'm not excited about here is the data url.

File: src/languages/wescheme/WeschemeParser.js (hunk 5 above)
Comment (schanzer) [inline src/languages/wescheme/WeschemeParser.js]: Stupid question - what IS the preferred style guide for whitepsace here?

Comment (pcardune) [inline src/languages/wescheme/WeschemeParser.js]: spaces around infix operators.

**Pull request merged**